lib/HttpClient.ts

```diff
@@ -241,18 +247,22 @@ export class HttpClient implements ifm.IHttpClient {
                 && this._allowRedirects
                 && redirectsRemaining > 0) {
 
-                const redirectUrl: any = response.message.headers["location"];
+                const redirectUrl: string | null = response.message.headers["location"];
                 if (!redirectUrl) {
                     // if there's no location to redirect to, we won't
                     break;
                 }
+                let parsedRedirectUrl = url.parse(redirectUrl);
+                if (parsedUrl.protocol == 'https:' && parsedUrl.protocol != parsedRedirectUrl.protocol && !this._allowRedirectDowngrade) {
+                    throw new Error("Redirect from HTTPS to HTTP protocol.");
```

Comment:
Can we make this error a little more explicit as to why this is bad and options to get around it if you're fine downgrading.

Something like `Redirect from HTTPS to HTTP protocol. This downgrade is not allowed for security reasons. If you want to allow this behavior, set the allowRedirectDowngrade option to true`

Reply:
Sure!

Reply:
```suggestion
                    throw new Error("Redirect from HTTPS to HTTP protocol. This downgrade is not allowed for security reasons. If you want to allow this behavior, set the allowRedirectDowngrade option to true.");
```

Reply:
I would simply say "Downgrading from HTTPS to HTTP is not permitted on a redirect.  If this is intended, set allowRedirectDowngrade to true"